internal/node/node_loader.js

```diff
@@ -206,51 +207,97 @@ function resolveRunfiles(...pathSegments) {
 
     maybe = loadAsDirectorySync(defaultPath);
     if (maybe) {
-      if (DEBUG) console.error('node_loader: resolved directory', maybe);
+      if (DEBUG) console.error('node_loader: resolved via directory', maybe);
       return maybe;
     }
   }
 
-  if (DEBUG) console.error('node_loader: resolved to default path', defaultPath);
   return defaultPath;
 }
 
-var originalResolveFilename = module.constructor._resolveFilename;
+const originalResolveFilename = module.constructor._resolveFilename;
 module.constructor._resolveFilename =
     function(request, parent) {
-  var failedResolutions = [];
-  var resolveLocations = [
-    request,
-    resolveRunfiles(request),
-    resolveRunfiles(
-        'TEMPLATED_user_workspace_name', 'TEMPLATED_label_package', 'node_modules', request),
-  ];
-  // Additional search path in case the build is across workspaces.
-  // See comment in node.bzl.
-  if ('TEMPLATED_label_workspace_name') {
-    resolveLocations.push(resolveRunfiles(
-        'TEMPLATED_label_workspace_name', 'TEMPLATED_label_package', 'node_modules', request));
+  if (DEBUG)
+    console.error(
+        `node_loader: resolve ${request} from ` +
+        `${parent && parent.filename ? parent.filename : ''}`);
+
+  const failedResolutions = [];
+
+  // Built-in modules, relative, absolute imports and npm dependencies
+  // can be resolved using request
+  try {
```

Comment:
this has lots of conditional logic, maybe we could write a unit test for this file?